hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -37,31 +37,132 @@
     /*
         CRUD interface for Trigger
      */
+
+    /**
+     * Create a new <code>Trigger</code>.  <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The new </code>Trigger</code> will be persisted.  When fully defined a call to
+     * {@link #updateTrigger(Trigger)}
+     * is needed to enable the </code>Trigger</code>.
+     * @param trigger
+     * @throws Exception If the </code>Trigger</code> already exists.
+     */
     void addTrigger(Trigger trigger) throws Exception;
+
+    /**
+     * The <code>Trigger</code> will be removed from the Alerts engine, as needed, and will no longer be persisted.
+     * @param triggerId
+     * @throws Exception
+     */
     void removeTrigger(String triggerId) throws Exception;
-    void updateTrigger(Trigger trigger) throws Exception;
-    Collection<Trigger> getTriggers() throws Exception;
+
+    /**
+     * Update the <code>Trigger</code>. <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The updated </code>Trigger</code> will be persisted.  If enabled the </code>Trigger</code>
+     * will be [re-]inserted into the Alerts engine and any prior dampening will be reset.
+     * @param triggerId
+     * @throws Exception If the </code>Trigger</code> does not exist.
+     */
+    Trigger updateTrigger(Trigger trigger) throws Exception;
+
     Trigger getTrigger(String triggerId) throws Exception;
 
+    Collection<Trigger> getAllTriggers() throws Exception;
+
```

Comment:
Also, as a general note for DefinitionsService: we should introduce also pagination helpers.
So, perhaps we should include in the API methods to getCountTrigger() or similar and getTriggers(from, to).
Also, we may need some kind of filtering/ordering (I don't know, but thinking loud about it).